pkgs/test/cc-wrapper/default.nix

```diff
@@ -17,36 +18,37 @@ in stdenv.mkDerivation {
 
     printf "checking whether compiler builds valid C binaries... " >&2
     $CC -o cc-check ${./cc-main.c}
-    ./cc-check
+    ${emulator} ./cc-check
 
     printf "checking whether compiler builds valid C++ binaries... " >&2
     $CXX -o cxx-check ${./cxx-main.cc}
-    ./cxx-check
+    ${emulator} ./cxx-check
 
     ${lib.optionalString (stdenv.isDarwin && stdenv.cc.isClang) ''
       printf "checking whether compiler can build with CoreFoundation.framework... " >&2
       mkdir -p foo/lib
       $CC -framework CoreFoundation -o core-foundation-check ${./core-foundation-main.c}
-      ./core-foundation-check
+      ${emulator} ./core-foundation-check
     ''}
 
 
     ${lib.optionalString (!stdenv.isDarwin) ''
       printf "checking whether compiler builds valid static C binaries... " >&2
       $CC ${staticLibc} -static -o cc-static ${./cc-main.c}
-      ./cc-static
-      ${lib.optionalString (stdenv.cc.isGNU && lib.versionAtLeast (lib.getVersion stdenv.cc.name) "8.0.0") ''
+      ${emulator} ./cc-static
+      ${lib.optionalString (stdenv.cc.isGNU && (stdenv.hostPlatform.isx86 || stdenv.hostPlatform.isAarch64)
```

Comment:
Why are platforms restricted here?

Reply:
I can't recall why I did that, maybe it was failing on riscv.